### ISSUE TITLE
Panick at 'attempt to add with overflow' after writing 512MB+ in 8MB chunks

### ISSUE THREAD

**Issue by nickelc** [stainless-steel/md5#11]
### Example
```rust
extern crate md5;

use std::io::prelude::*;

fn main() {
    let data = vec![0; 8 * 1024 * 1024];
    let mut ctx = md5::Context::new();
    let mut len = 0;
    let mut count = 0;
    loop {
        match ctx.write(&data) {
            Ok(n) => {
                len += n;
                count += 1;
            }
            Err(e) => {
                println!("{}", e);
                break;
            }
        }
        println!("#{} written: {}", count, len);
    }
}
```
### Stacktrace
```
#61 written: 511705088
#62 written: 520093696
#63 written: 528482304
thread 'main' panicked at 'attempt to add with overflow', src/lib.rs:118:12
stack backtrace:
   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
             at libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
   1: std::sys_common::backtrace::print
             at libstd/sys_common/backtrace.rs:71
             at libstd/sys_common/backtrace.rs:59
   2: std::panicking::default_hook::{{closure}}
             at libstd/panicking.rs:211
   3: std::panicking::default_hook
             at libstd/panicking.rs:227
   4: std::panicking::rust_panic_with_hook
             at libstd/panicking.rs:475
   5: std::panicking::continue_panic_fmt
             at libstd/panicking.rs:390
   6: rust_begin_unwind
             at libstd/panicking.rs:325
   7: core::panicking::panic_fmt
             at libcore/panicking.rs:77
   8: core::panicking::panic
             at libcore/panicking.rs:52
   9: md5::Context::consume
             at src/lib.rs:118
  10: <md5::Context as std::io::Write>::write
             at ./src/lib.rs:195
  11: test::main
             at examples/test.rs:11
```

**Comment by IvanUkhov** [comment]
Thanks a lot for reporting this!

**Comment by IvanUkhov** [comment]
Fixed in #12.
